plugins/modules/zypper_repository_info.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Tobias Zeuch <tobias.zeuch@sap.com>
+#
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: zypper_repository_info
+author: "Tobias Zeuch"
+version_added: 9.4.0
+short_description: List Zypper repositories
+description:
+    - List Zypper repositories on SUSE and openSUSE.
+    - Note: for info about packages, use the packages ansible.builtin.package_facts
+extends_documentation_fragment:
+    - community.general.attributes
+    - community.general.attributes.info_module
+
+requirements:
+    - "zypper >= 1.0  (included in openSUSE >= 11.1 or SUSE Linux Enterprise Server/Desktop >= 11.0)"
+    - python-xml
+notes:
+    - "for info about packages, use the module M(ansible.builtin.package_facts)"
+'''
+
+EXAMPLES = '''
+- name: List registered repositories and store in variable repositories
+  community.general.zypper_repository_info:
+  register: repodatalist
+'''
+
+RETURN = r'''
+repodatalist:
+    description:
+        - a list of repository descriptions like it is returned by the command zypper repos
+    type: list
+    elements: dict
+    contains:
+        alias:
+            description: the alias of the repository
+            type: str
+        autorefresh:
+            description: indicates, if autorefresh is enabled on the repository
+            type: int
+        enabled:
+            description: indicates, if the repository is enabled
+            type: int
+        gpgcheck:
+            description: indicates, if the GPG signature of the repository meta data is checked
+            type: int
+        name:
+            description: the name of the repository
+            type: str
+        priority:
+            description: the priority of the repository
+            type: int
+        url:
+            description: the url of the repository on the internet
+            type:str
+        sample:
```

Comment:
This needs to be one level up, since it's not another key for the `repodatalist` return value:
```suggestion
    sample:
```